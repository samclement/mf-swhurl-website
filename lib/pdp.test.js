const parser = require('./pdp.js')
const fs = require('fs')
const path = require('path')
const html = fs.readFileSync(path.resolve('./test-data/pdp.html'), 'utf8')

describe('PLP', () => {
  it('Should parse the navigation', () => {
    expect(parser.parseProduct(html)).toMatchSnapshot()
  })
})
