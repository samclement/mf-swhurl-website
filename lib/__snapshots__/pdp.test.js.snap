// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PLP Should parse the navigation 1`] = `
Object {
  "outfit_1269352": Object {
    "products": Array [
      Object {
        "product": Object {
          "code": "1269352",
          "designer": Object {
            "name": "Track &amp; Bliss",
            "url": "",
          },
          "image": Object {
            "large": "//assetsprx.matchesfashion.com/img/product/1269352_1_large.jpg",
            "medium": "//assetsprx.matchesfashion.com/img/product/1269352_1_medium.jpg",
            "thumbnail": "//assetsprx.matchesfashion.com/img/product/1269352_1_thumbnail.jpg",
          },
          "image.altText": "Track &amp; Bliss The Knot cotton tank",
          "isMyStylistOnly": false,
          "isOneSize": false,
          "name": "The Knot cotton tank",
          "priceData": Object {
            "formattedValue": "£58",
          },
          "slug": "Activewear",
          "url": "/products/Track-%26-Bliss-The-Knot-cotton-tank-1269352",
          "variantOptions": Array [
            Object {
              "code": "1269352000003",
              "comingSoon": "",
              "qty": "12",
              "sizeData": "XS",
              "sizeDataBaseCode": "10002",
              "sizeDataCode": "20007",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1269352000004",
              "comingSoon": "",
              "qty": "13",
              "sizeData": "S",
              "sizeDataBaseCode": "10003",
              "sizeDataCode": "20008",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1269352000005",
              "comingSoon": "",
              "qty": "10",
              "sizeData": "M",
              "sizeDataBaseCode": "10004",
              "sizeDataCode": "20009",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1269352000006",
              "comingSoon": "",
              "qty": "4",
              "sizeData": "L",
              "sizeDataBaseCode": "10005",
              "sizeDataCode": "20010",
              "stockLevelCode": "inStock",
            },
          ],
        },
      },
      Object {
        "product": Object {
          "code": "1269356",
          "designer": Object {
            "name": "Track &amp; Bliss",
            "url": "",
          },
          "image": Object {
            "large": "//assetsprx.matchesfashion.com/img/product/1269356_1_large.jpg",
            "medium": "//assetsprx.matchesfashion.com/img/product/1269356_1_medium.jpg",
            "thumbnail": "//assetsprx.matchesfashion.com/img/product/1269356_1_thumbnail.jpg",
          },
          "image.altText": "Track &amp; Bliss Skywalker reversible sports bra",
          "isMyStylistOnly": false,
          "isOneSize": false,
          "name": "Skywalker reversible sports bra",
          "priceData": Object {
            "formattedValue": "£60",
          },
          "slug": "Activewear",
          "url": "/products/Track-%26-Bliss-Skywalker-reversible-sports-bra-1269356",
          "variantOptions": Array [
            Object {
              "code": "1269356000003",
              "comingSoon": "",
              "qty": "3",
              "sizeData": "XS",
              "sizeDataBaseCode": "10002",
              "sizeDataCode": "20007",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1269356000006",
              "comingSoon": "",
              "qty": "1",
              "sizeData": "L",
              "sizeDataBaseCode": "10005",
              "sizeDataCode": "20010",
              "stockLevelCode": "lowStock",
            },
            Object {
              "code": "1269356000004",
              "comingSoon": "",
              "qty": "9",
              "sizeData": "S",
              "sizeDataBaseCode": "10003",
              "sizeDataCode": "20008",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1269356000005",
              "comingSoon": "",
              "qty": "13",
              "sizeData": "M",
              "sizeDataBaseCode": "10004",
              "sizeDataCode": "20009",
              "stockLevelCode": "inStock",
            },
          ],
        },
      },
      Object {
        "product": Object {
          "code": "1218010",
          "designer": Object {
            "name": "Adidas By Stella McCartney",
            "url": "",
          },
          "image": Object {
            "large": "//assetsprx.matchesfashion.com/img/product/1218010_1_large.jpg",
            "medium": "//assetsprx.matchesfashion.com/img/product/1218010_1_medium.jpg",
            "thumbnail": "//assetsprx.matchesfashion.com/img/product/1218010_1_thumbnail.jpg",
          },
          "image.altText": "Adidas By Stella McCartney Performance Essentials leggings",
          "isMyStylistOnly": false,
          "isOneSize": false,
          "name": "Performance Essentials leggings",
          "priceData": Object {
            "formattedValue": "£63",
          },
          "slug": "Activewear",
          "url": "/products/Adidas-By-Stella-McCartney-Performance-Essentials-leggings-1218010",
          "variantOptions": Array [
            Object {
              "code": "1218010000002",
              "comingSoon": "",
              "qty": "1",
              "sizeData": "XXS",
              "sizeDataBaseCode": "10001",
              "sizeDataCode": "20006",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1218010000003",
              "comingSoon": "",
              "qty": "27",
              "sizeData": "XS",
              "sizeDataBaseCode": "10002",
              "sizeDataCode": "20007",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1218010000004",
              "comingSoon": "",
              "qty": "56",
              "sizeData": "S",
              "sizeDataBaseCode": "10003",
              "sizeDataCode": "20008",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1218010000005",
              "comingSoon": "",
              "qty": "24",
              "sizeData": "M",
              "sizeDataBaseCode": "10004",
              "sizeDataCode": "20009",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1218010000006",
              "comingSoon": "",
              "qty": "16",
              "sizeData": "L",
              "sizeDataBaseCode": "10005",
              "sizeDataCode": "20010",
              "stockLevelCode": "inStock",
            },
          ],
        },
      },
      Object {
        "product": Object {
          "code": "1256969",
          "designer": Object {
            "name": "Adidas By Stella McCartney",
            "url": "",
          },
          "image": Object {
            "large": "//assetsprx.matchesfashion.com/img/product/1256969_1_large.jpg",
            "medium": "//assetsprx.matchesfashion.com/img/product/1256969_1_medium.jpg",
            "thumbnail": "//assetsprx.matchesfashion.com/img/product/1256969_1_thumbnail.jpg",
          },
          "image.altText": "Adidas By Stella McCartney Ultraboost X low-top mesh trainers",
          "isMyStylistOnly": false,
          "isOneSize": false,
          "name": "Ultraboost X low-top mesh trainers",
          "priceData": Object {
            "formattedValue": "£200",
          },
          "slug": "",
          "url": "/products/Adidas-By-Stella-McCartney-Ultraboost-X-low-top-mesh-trainers-1256969",
          "variantOptions": Array [
            Object {
              "code": "1256969000006",
              "comingSoon": "",
              "qty": "8",
              "sizeData": "3.5 UK",
              "sizeDataBaseCode": "10017",
              "sizeDataCode": "20328",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000007",
              "comingSoon": "",
              "qty": "16",
              "sizeData": "4 UK",
              "sizeDataBaseCode": "10018",
              "sizeDataCode": "20329",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000008",
              "comingSoon": "",
              "qty": "18",
              "sizeData": "4.5 UK",
              "sizeDataBaseCode": "10019",
              "sizeDataCode": "20330",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000009",
              "comingSoon": "",
              "qty": "14",
              "sizeData": "5 UK",
              "sizeDataBaseCode": "10020",
              "sizeDataCode": "20331",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000010",
              "comingSoon": "",
              "qty": "24",
              "sizeData": "5.5 UK",
              "sizeDataBaseCode": "10021",
              "sizeDataCode": "20316",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000011",
              "comingSoon": "",
              "qty": "31",
              "sizeData": "6 UK",
              "sizeDataBaseCode": "10022",
              "sizeDataCode": "20317",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000012",
              "comingSoon": "",
              "qty": "9",
              "sizeData": "6.5 UK",
              "sizeDataBaseCode": "10023",
              "sizeDataCode": "20318",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000013",
              "comingSoon": "",
              "qty": "6",
              "sizeData": "7 UK",
              "sizeDataBaseCode": "10024",
              "sizeDataCode": "20319",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000014",
              "comingSoon": "",
              "qty": "13",
              "sizeData": "7.5 UK",
              "sizeDataBaseCode": "10025",
              "sizeDataCode": "20320",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000015",
              "comingSoon": "",
              "qty": "7",
              "sizeData": "8 UK",
              "sizeDataBaseCode": "10026",
              "sizeDataCode": "20321",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000016",
              "comingSoon": "",
              "qty": "1",
              "sizeData": "8.5 UK",
              "sizeDataBaseCode": "10124",
              "sizeDataCode": "20322",
              "stockLevelCode": "inStock",
            },
            Object {
              "code": "1256969000017",
              "comingSoon": "",
              "qty": "1",
              "sizeData": "9 UK",
              "sizeDataBaseCode": "10125",
              "sizeDataCode": "20323",
              "stockLevelCode": "inStock",
            },
          ],
        },
      },
    ],
  },
}
`;
